### ISSUE TITLE
posibillity to use upstream kernel and uboot on R3?

### ISSUE THREAD

**Issue by runborg** [frank-w/BPI-Router-Images#1]
Hi @frank-w!

This is not directly an issue with this repository, bot more about questions to these boards in general!

I'm trying to build an image of [VyOS](https://vyos.io/) running on the BananaPi R3 board, doing this i cant use this repository directly, but are trying to use it as a base for generating the image files needed. My image needs a quite "special" boot sequence to get going using their official tools (uboot with efi and grub).

In that regards i'm wondering on your experience with running upstream u-boot and newer vanilla kernel on these boards?
Have you tried it, will it boot at all? known things not working etc.?

In advance thanks for an great debian/ubuntu build package for the BananaPi R* boards!
cheers

**Comment by frank-w** [comment]
Hi it depends on which kernel-version you need. Last LTS (6.1) misses lots of things. E.g. thermal is merged in upcoming 6.5. for other look into my main-branches for differences between mainline and my kernel

Why can't you use my kernelsource? This repo uses already built kernel with specific packed structure. If you use mainline kernel you need to compile kernel yourself and change the install function (and skip kernel download).

**Comment by runborg** [comment]
Thnx for a quick reply @frank-w ! :)

Was looking into 6.4, i see you've created a 6.4-main branch, is this branch working on R3 at this point in time?

About compiling the kernel myself i will need to do that anyways as vyos has quite a lot of defconfig customisation to get everything up'n'running the way its supposed to be .. :)  also to get their upgrade scripts working the everything inkl. the kernel needs to be packaged in quite a special way..

**Comment by frank-w** [comment]
Yes 6.4 works with bpi-r3 and should include my last state of support,but have not tested all...

And yes the build-system of thos repo creates bootable images for all 4 boards i have here...maybe it needs some userspace customization i'm currently working on

**Comment by mrninhvn** [comment]
Hi @runborg 
Have you been successful with VyOs yet?
